dm/worker/subtask.go

```diff
@@ -457,8 +473,12 @@ func (st *SubTask) Close() {
 	updateTaskState(st.cfg.Name, st.cfg.SourceID, pb.Stage_Stopped)
 }
 
-// Pause pauses the running sub task.
+// Pause pauses a running sub task or a sub task paused by error.
 func (st *SubTask) Pause() error {
+	if st.markResultCanceled() {
```

Comment:
if `markResultCanceled` is called before chaning stage, user need to pause twice to pin the paused status. I think his expectation is pausing once

Reply:
+1

Reply:
If user pause a running task, DM will [callCurrentCancel](https://github.com/pingcap/dm/blob/068c30eb344c2556f13b1dbf7a6e55feed7ca1f3/dm/worker/subtask.go#L466) and [fetchResult](https://github.com/pingcap/dm/blob/068c30eb344c2556f13b1dbf7a6e55feed7ca1f3/dm/worker/subtask.go#L252) will mark the task paused and set isCanceled to true.
If task paused by error, isCanceled is false, so simply replace the value, and [task checker](https://github.com/pingcap/dm/blob/068c30eb344c2556f13b1dbf7a6e55feed7ca1f3/dm/worker/task_checker.go#L260) will not resume it.

Reply:
so the discription of https://github.com/pingcap/dm/issues/1729 is wrong? task will keep paused when user manually paused it, but just before manually pause, some error would occur and get a paused status, so we wrongly take it as manually pause didn't take effect?

Reply:
Yes！